core/modules/replica/Controller.h

```diff
@@ -27,22 +27,174 @@
  */
 
 // System headers
+#include <functional>
 #include <map>
 #include <memory>
 #include <thread>
 #include <vector>
 
 // Qserv headers
 #include "replica/Request.h"
-#include "replica/RequestTypesFwd.h"
 #include "replica/ServiceProvider.h"
 #include "util/Mutex.h"
 
 // Forward declarations
 namespace lsst {
 namespace qserv {
 namespace replica {
-    class ControllerImpl;
+
+    class ReplicationRequest;
+    class DeleteRequest;
+
+    typedef std::shared_ptr<ReplicationRequest> ReplicationRequestPtr;
+    typedef std::shared_ptr<DeleteRequest>      DeleteRequestPtr;
+
+    class FindRequest;
+    class FindAllRequest;
+
+    typedef std::shared_ptr<FindRequest>    FindRequestPtr;
+    typedef std::shared_ptr<FindAllRequest> FindAllRequestPtr;
+
+    class EchoRequest;
+
+    typedef std::shared_ptr<EchoRequest> EchoRequestPtr;
+
+    class IndexRequest;
+
+    typedef std::shared_ptr<IndexRequest> IndexRequestPtr;
+
+    class SqlQueryRequest;
+    class SqlCreateDbRequest;
+    class SqlDeleteDbRequest;
+    class SqlEnableDbRequest;
+    class SqlDisableDbRequest;
+    class SqlGrantAccessRequest;
+    class SqlCreateTableRequest;
+    class SqlDeleteTableRequest;
+    class SqlRemoveTablePartitionsRequest;
+    class SqlDeleteTablePartitionRequest;
+
+    typedef std::shared_ptr<SqlQueryRequest>                 SqlQueryRequestPtr;
+    typedef std::shared_ptr<SqlCreateDbRequest>              SqlCreateDbRequestPtr;
+    typedef std::shared_ptr<SqlDeleteDbRequest>              SqlDeleteDbRequestPtr;
+    typedef std::shared_ptr<SqlEnableDbRequest>              SqlEnableDbRequestPtr;
+    typedef std::shared_ptr<SqlDisableDbRequest>             SqlDisableDbRequestPtr;
+    typedef std::shared_ptr<SqlGrantAccessRequest>           SqlGrantAccessRequestPtr;
+    typedef std::shared_ptr<SqlCreateTableRequest>           SqlCreateTableRequestPtr;
+    typedef std::shared_ptr<SqlDeleteTableRequest>           SqlDeleteTableRequestPtr;
+    typedef std::shared_ptr<SqlRemoveTablePartitionsRequest> SqlRemoveTablePartitionsRequestPtr;
+    typedef std::shared_ptr<SqlDeleteTablePartitionRequest>  SqlDeleteTablePartitionRequestPtr;
+
+    class StopReplicationRequestPolicy;
+    class StopDeleteRequestPolicy;
+    class StopFindRequestPolicy;
+    class StopFindAllRequestPolicy;
+    class StopEchoRequestPolicy;
+    class StopIndexRequestPolicy;
+    class StopSqlRequestPolicy;
```

Comment:
I guess this is the group of Stop Policies, and so forth.

Reply:
Correct! These forward declarations are used as template parameters of:
```
    template <typename POLICY> class StopRequest;
```